cdf/cdf.js

```diff
@@ -0,0 +1,143 @@
+const { interpolate } = require('./functions');
+
+class Cdf {
+  /**
+   * @param {number[]} xs
+   * @param {number[]} ys
+   */
+  constructor(xs, ys) {
+    this.xs = xs;
+    this.ys = ys;
+
+    if (!this.validateSize(this.xs, this.ys)) {
+      const corrected = this.convertToNewLength(this.xs.length);
+      this.xs = corrected.xs;
+      this.ys = corrected.ys;
+    }
+
+    const sorted = this.order(this.xs, this.ys);
+    this.xs = sorted.xs;
+    this.ys = sorted.ys;
+  }
+
+  /**
+   * Order them to make sure that xs are increasing
+   * @param {number[]} xs
+   * @param {number[]} ys
+   * @return {{ys: number[], xs: number[]}}
+   */
+  order(xs, ys) {
+    const xsYs = xs.map((v, i) => ({ ys: ys[i], xs: v }));
+    const sorted = xsYs.sort((a, b) => {
+      if (a.xs > b.xs) return 1;
+      if (a.xs < b.xs) return -1;
+      return 0;
+    });
+
+    const XS = sorted.map(v => v.xs);
+    const YS = sorted.map(v => v.ys);
+
+    return { xs: XS, ys: YS };
+  }
```

Comment:
For these to be valid CDFs, when Xs are sorted, Ys should also be sorted (only being equal or going up). Can you write another check to make sure that is the case?

Reply:
Yes, sure, going to add tests later in another issue